xtokens/src/lib.rs

```diff
@@ -552,6 +561,18 @@ pub mod module {
 						Error::<T>::DistinctReserveForAssetAndFee
 					);
 				}
+
+				// per asset check
+				let amount = match asset.fun {
+					Fungibility::Fungible(amount) => amount,
+					Fungibility::NonFungible(_) => 1,
+				};
+
+				let rate_limiter_id = T::RateLimiterId::get();
+
+				// try consume quota of the rate limiter.
+				T::RateLimiter::try_consume(rate_limiter_id, asset.id, amount, Some(who.clone()))
```

Comment:
```suggestion
				T::RateLimiter::try_consume(rate_limiter_id, asset.id, amount, Some(&who))
```

I think this should work?